shopelectro/tasks.py

```diff
@@ -61,18 +63,30 @@ def update_catalog():
         collect_static()
     ]
 
-# @todo #690:60m Handle errors in check_purchase.
-#  Report failed attempts. Schedule it in the celery beat.
+# @todo #690:30m Schedule check_purchase in the celery beat.
```

Comment:
@artemiy312 , i already suggested asincio instead of celery beat at the parent issue's PR:
https://github.com/fidals/shopelectro/pull/695#discussion_r249236197
What do you think about it?

Reply:
I believe it is bad idea:
- Django is a synchronous framework that will interfere with asynchronous code
- We need a stateful pipeline for message delivery to be able to save messages and process them later in case of deployment, bad internet connection, hardware problems, any other thing that prevent a message processing
- We already have working solution, that looks fine and aren't annoying us

Reply:
We can move into `rq`, `huye` or something else, but it seems it will be the same picture

Reply:
i'll answer just for history:
- it's not big deal, i think. We can our own async utils process it with our own separated module
- we can provide safety by hands in every concrete case. In orders case we can (and we do) put them to the db synchronously and then do some dangerous async operations
- we had problems with celery not long time ago: unsent mails, not launched celery services and so on. Problems for example: [one](https://github.com/fidals/shopelectro/issues/585), [two](https://github.com/fidals/shopelectro/issues/561), [three](https://github.com/fidals/shopelectro/issues/424)